tests/trio109.py

```diff
@@ -0,0 +1,150 @@
+# split into two files for ease of debugging
+import trio
+
+
+async def bar():
+    await bar()
+
+
+async def foo1(delay, to):
+    await bar()
+    yield 5
+    await bar()
+
+
+async def foo2():
+    yield  # safe
+    yield  # error
+    await bar()
+
+
+async def foo3():  # error
+    await bar()
+    yield
+
+
+async def foo4():
+    yield 1
+    await (yield 1)  # error
+
+
+# async with
+async def foo_async_with():  # safe, checkpoint on both enter and exit of async with
```

Comment:
Nooooot *always* the case; Trio promises that at least one of enter/exit will be checkpoints, but it might not be both.

I think we should "handle" this by documenting it as a limitation of the analysis and moving on, at least until we're otherwise finished.

Reply:
Fully handled :+1: